ca/config.go

```diff
@@ -415,8 +415,8 @@ func LoadTLSCreds(rootCA RootCA, paths CertPaths) (*MutableTLSCreds, *MutableTLS
 		}
 
 		keyPair, newErr = tls.X509KeyPair(cert, key)
-		if err != nil {
-			return nil, nil, err
+		if newErr != nil {
+			return nil, nil, newErr
```

Comment:
Return `err` here like above. We should return the original error if we try using the temporary key and it doesn't work either. Good catch on the `if` statement though.